### PR TITLE
Fix panic in discoverynamespaces.go

### PR DESCRIPTION
**Pull request by dwj300**
**Please provide a description of this PR:**
Fixes https://github.com/istio/istio/issues/36014

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/filter/discoverynamespaces.go [hunk 1]
```diff
@@ -71,8 +71,14 @@ func (d *discoveryNamespacesFilter) Filter(obj interface{}) bool {
 		return true
 	}
 
+	// When an object is deleted, obj could be a DeletionFinalStateUnknown marker item.
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/filter/discoverynamespaces.go:74]: Can you process `DeletionFinalStateUnknown` as well, so we donot miss a delete event

Comment (dwj300) [reply pilot/pkg/serviceregistry/kube/controller/filter/discoverynamespaces.go:75]: good call - done

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@dwj300  please sign the cla

**Review by howardjohn — APPROVED** [review]

**Comment by dwj300** [comment]
> @dwj300 please sign the cla

done! Can you re-run the CLA scan?

**Pull request merged**